Add Events page section for Observer run_if functionality (#2438)
bevyengine/bevy-website: content/learn/book/control-flow/events.md

[`Component`]: https://docs.rs/bevy/latest/bevy/prelude/trait.Component.html

### Conditionally Triggering Observers

> The [Skipping Systems page] introduced conditional checks that can be used to control when a system should run.

Reviewer: ```suggestion
The [Skipping Systems page] introduced run conditions, which can be used to control when a system should run.
```

> Observers can also be controlled with conditional checks using a similar [`.run_if`] pattern.

Reviewer: ```suggestion
These work for observers too!
```
